com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/security/HanaSecureTokenStore.java

```diff
@@ -164,8 +163,7 @@ private OAuth2AccessToken readAccessTokenFromSecureStore(String tokenKey) {
             throw new IllegalArgumentException(Messages.TOKEN_NOT_FOUND_IN_SECURE_STORE);
         }
         byte[] decompressedBytes = CompressUtil.decompress(tokenBytes);
-        OAuth2AccessToken accessToken = new TokenFactory().createToken(new String(decompressedBytes, StandardCharsets.UTF_8));
-        return accessToken;
+        return SerializationUtils.deserialize(decompressedBytes);
```

Comment:
What would happen if there are tokens stored as base 64 encoded strings in the token store? Would the deserialization fail here?